bloptools/experiments/tests/__init__.py

```diff
@@ -1,20 +1,34 @@
 import bluesky.plan_stubs as bps
 import numpy as np
-from ophyd import Signal
+from ophyd import Component as Cpt
+from ophyd import Device, Signal
 
 
 def get_dofs(n=2):
     return [Signal(name=f"x{i+1}", value=0) for i in range(n)]
 
 
+def get_device(name="dofs", n=2):
+    components = {}
+
+    for i in range(n):
+        components[f"{i+1}"] = Cpt(Signal, value=i + 1)
+
+    cls = type("DOF", (Device,), components)
+
+    device = cls(name=name)
+
+    return [getattr(device, attr) for attr in device.read_attrs]
```

Comment:
Thanks for implementing it! Why not return the generated device?
Also, is it used anywhere? I see only the `get_dofs()` is used. I think it's fine if you only need a list of individual signals.

Reply:
The optimizer uses an iterable of signals (so that we can optimize many devices), so having them as one device just makes it a bit more complicated for no reason

Reply:
Let's keep it for now. It's good to have a quick way to use `Device`-based components if needed in the future.